target/linux/ar71xx/files/arch/mips/ath79/mach-fritz1750e.c

```diff
@@ -0,0 +1,252 @@
+/*
+ *  AVM FRITZ!WLAN Repeater 1750E board support
+ *
+ *  Copyright (C) 2018 David Bauer <mail@david-bauer.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 as published
+ *  by the Free Software Foundation.
+ */
+
+#include <linux/init.h>
+#include <linux/ath9k_platform.h>
+#include <linux/etherdevice.h>
+#include <linux/gpio.h>
+#include <linux/platform_device.h>
+
+#include <linux/mtd/mtd.h>
+#include <linux/mtd/partitions.h>
+#include <linux/platform_data/mdio-gpio.h>
+#include <linux/platform_data/phy-at803x.h>
+#include <linux/spi/spi_gpio.h>
+#include <linux/spi/74x164.h>
+
+#include <asm/mach-ath79/ath79.h>
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-spi.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+
+
+#define FRITZ1750E_GPIO_SHIFT_SER	15   /* DS,   Data Serial Input */
+#define FRITZ1750E_GPIO_SHIFT_SRCLK	14 /* SHCP, Shift Reg Clock Input */
+
+#define FRITZ1750E_SSR_BIT_0			0
+#define FRITZ1750E_SSR_BIT_1			1
+#define FRITZ1750E_SSR_BIT_2			2
+#define FRITZ1750E_SSR_BIT_3			3
+#define FRITZ1750E_SSR_BIT_4			4
+#define FRITZ1750E_SSR_BIT_5			5
+#define FRITZ1750E_SSR_BIT_6			6
+#define FRITZ1750E_SSR_BIT_7			7
+
+#define FRITZ1750E_74HC_GPIO_BASE		32
+#define FRITZ1750E_74HC_GPIO_LED_RSSI0	(FRITZ1750E_74HC_GPIO_BASE + 0)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI1	(FRITZ1750E_74HC_GPIO_BASE + 1)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI2		(FRITZ1750E_74HC_GPIO_BASE + 2)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI3		(FRITZ1750E_74HC_GPIO_BASE + 3)
+#define FRITZ1750E_74HC_GPIO_LED_RSSI4		(FRITZ1750E_74HC_GPIO_BASE + 4)
+#define FRITZ1750E_74HC_GPIO_LED_WLAN		  (FRITZ1750E_74HC_GPIO_BASE + 5)
+#define FRITZ1750E_74HC_GPIO_LED_POWER		  (FRITZ1750E_74HC_GPIO_BASE + 6)
+
```

Comment:
lots of indenting and space/vs tabs errors in this file. please make it all nice and consistent.